modules/packages/Socket.chpl

```diff
@@ -242,9 +241,24 @@ proc ipAddr.writeThis(f) throws {
 }
 
 pragma "no doc"
-operator timeval.=(other: real) {
-  this.tv_sec = other:c_long;
-  this.tv_usec = (other - this.tv_sec) * 1000000;
+operator struct_timeval.=(other: real) {
+  this.tv_sec = (other:c_long):time_t;
+  this.tv_usec = ((other - this.tv_sec) * 1000000):suseconds_t;
+}
+
+/*
+  Get a :type:`~POSIX.struct_timeval` set for indefinite timeout.
+
+  This is the default value used in various procedures in this module
+
+  - `tv_sec` is assigned a value of ``-1``
+  - `tv_usec` is assigned a value of ``0``
+*/
+proc indefiniteTimeout {
+  var dtv : struct_timeval;
+  dtv.tv_sec = (-1):c_int:time_t;
+  dtv.tv_usec = 0:c_int:suseconds_t;
+  return dtv;
```

Comment:
Alternatively, it seems as though you might be able to simply write this as:

```chapel
const indefiniteTimeout = new struct_timeval(tv_sec=..., tv_usec=...);
```

(or potentially using split initialization if you wanted to hide the details of its initial value?)

```chapel
const indefiniteTimeout: struct_timeval;
indefiniteTimout = new struct_timeval(tv_sec=..., tv_usec=...);
```